api/go.mod

```diff
@@ -3,47 +3,78 @@ module github.com/kiegroup/kogito-serverless-operator/api
 go 1.19
 
 require (
+	github.com/onsi/ginkgo/v2 v2.9.5
+	github.com/onsi/gomega v1.27.7
 	github.com/serverlessworkflow/sdk-go/v2 v2.2.3
 	k8s.io/api v0.27.2
 	k8s.io/apimachinery v0.27.2
+	k8s.io/client-go v0.27.2
```

Comment:
Why are you using the client go in the API module? We should not fetch anything from there. It's model/types and related use cases only.